python/lsst/ts/mtreflector/csc.py

```diff
@@ -0,0 +1,231 @@
+# This file is part of ts_mtreflector.
+#
+# Developed for the Vera C. Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["MTReflectorCsc", "run_mtreflector"]
+
+import asyncio
+import types
+from typing import Any
+
+from lsst.ts import salobj
+from lsst.ts.xml.enums.MTReflector import MTReflectorStatus
+
+from . import __version__
+from .config_schema import CONFIG_SCHEMA
+from .controller import COMMUNICATION_TIMEOUT, MTReflectorController
+
+
+class MTReflectorCsc(salobj.ConfigurableCsc):
+    """MTReflector controller
+
+    Parameters
+    ----------
+    config_dir : `None` or `str`, optional
+        Directory of configuration files, or None for the standard
+        configuration directory (obtained from `_get_default_config_dir`).
+        This is provided for unit testing.
+    initial_state : `State` or `int`, optional
+        The initial state of the CSC. This is provided for unit testing,
+        as real CSCs should start up in `State.STANDBY`, the default.
+    override : `str`, optional
+        Configuration override file to apply if ``initial_state`` is
+        `State.DISABLED` or `State.ENABLED`.
+    simulation_mode : `bool`, optional
+        Simulation mode; one of:
+
+        * False for normal operation
+        * True for simulation
+
+    Attributes
+    ----------
+    mtreflector_controller : `MTReflectorController`
+        The controller representing the labjack controller
+        which is controlling the reflector.
+    """
+
+    version = __version__
+    valid_simulation_modes = (0, 1)
+
+    def __init__(
+        self,
+        config_dir: None | str = None,
+        initial_state: salobj.State = salobj.State.STANDBY,
+        override: str = "",
+        simulation_mode: bool = False,
+    ) -> None:
+        self.reflector_controller = None
```

Comment:
I am surprised that mypy allows this. Please add a type here so it is clear what this attribute represents.

Reply:
I still don't see the type here.

Reply:
It does have a type.